be/src/storage/rowset_merger.cpp

```diff
@@ -271,13 +274,18 @@ class RowsetMergerImpl : public RowsetMerger {
             if (res.value().empty()) {
                 entry.segment_itr = new_empty_iterator(schema, _chunk_size);
             } else {
-                entry.segment_itr = new_heap_merge_iterator(res.value());
+                entry.segment_itr = std::move(new_heap_merge_iterator(res.value()));
```

Comment:
> Is std::move needed? return value is already a rvalue ref.

Maybe it is not necessary, but it can reduce the construction of `shared_ptr`. So i think use `move` is ok.